src/libjasper/jpc/jpc_t2dec.c

```diff
@@ -195,10 +195,21 @@ static int jpc_dec_decodepkt(jpc_dec_t *dec, jas_stream_t *pkthdrstream, jas_str
 			}
 			if (jpc_ms_gettype(ms) != JPC_MS_SOP) {
 				jpc_ms_destroy(ms);
-				jas_eprintf("missing SOP marker segment\n");
+				jas_eprintf("cannot get (SOP) marker segment\n");
+				return -1;
+			}
+			/* checking the packet sequence number */
+			if (dec->numpkts % (UINT_FAST16_MAX + 1) != ms->parms.sop.seqno) {
```

Comment:
This is a division by zero, because `UINT_FAST16_MAX+1` is an integer overflow which is `0`.

Reply:
Oops. My bad. I'll fix this zero division ASAP.

Reply:
When you fix this, please add a comment explaining the formula - I have no idea why this seqno is expected.

Reply:
I'll add the formula by amending this commit.